packages/server/test/profile.integration.ts

```diff
@@ -14,6 +14,107 @@ import { DesktopNotifModel } from 'notification/desktop-notif.entity';
 describe('Profile Integration', () => {
   const supertest = setupIntegrationTest(ProfileModule);
 
+  describe('POST /profile/:id/edit_user', () => {
+    it('Does not allow unauthorized edit', async () => {
+      await UserFactory.create();
+
+      await supertest()
+        .post(`/profile/1/edit_user`)
+        .send({
+          email: 'hello@world.com',
+          first_name: 'bob',
+          password: 'onion',
+          last_name: 'chungus',
+          selected_course: 134,
+          sid: 123,
+          photo_url: '',
+          courses: [],
+        })
+        .expect(401);
+    });
+
+    it('Cannot edit user that does not exist', async () => {
+      const user = await UserFactory.create();
+
+      await supertest({ userId: user.id })
+        .post(`/profile/0/edit_user`)
+        .send({
+          email: 'hello@world.com',
+          first_name: 'bob',
+          password: 'onion',
+          last_name: 'chungus',
+          selected_course: 134,
+          sid: 123,
+          photo_url: '',
+          courses: [],
+        })
+        .expect(404);
+    });
+
+    it('edits user', async () => {
+      const user = await UserFactory.create();
+
+      await supertest({ userId: user.id })
+        .post(`/profile/${user.id}/edit_user`)
+        .send({
+          email: 'hello@world.com',
+          first_name: 'bob',
+          password: 'onion',
+          last_name: 'chungus',
+          selected_course: 134,
```

Comment:
Are we not checking if selected courses exist on this endpoint?

Reply:
The endpoint takes an object of type `UBCuserParam` which has a field for selected course. But it edits the user itself and does nothing with the course value.